### PR TITLE
refactor(metrics): use absolute gauge values rather than deltas (v5.0)

### PR DESCRIPTION
**Pull request by thalesmg**
https://emqx.atlassian.net/browse/EMQX-8548

Currently, we face several issues trying to keep resource metrics
reasonable.  For example, when a resource is re-created and has its
metrics reset, but then its durable queue resumes its previous work
and leads to strange (often negative) metrics.

Instead using `counters` that are shared by more than one worker to
manage gauges, we introduce an ETS table whose key is not only scoped
by the Resource ID as before, but also by the worker ID.  This way,
when a worker starts/terminates, they should set their own gauges to
their values (often 0 or `replayq:count` when resuming off a queue).
With this scoping and initialization procedure, we'll hopefully avoid
hitting those strange metrics scenarios and have better control over
the gauges.

### CODE REVIEW

**Comment by thalesmg** [comment]
Requires https://github.com/kafka4beam/wolff/pull/41 to have proper metrics for the Kafka Producer bridge.

**Review by ieQu1 — DISMISSED** [review]

**Comment by thalesmg** [comment]
@ieQu1 sorry, forgot to update [this clause](https://github.com/thalesmg/emqx/blob/77feb8366a99d252496875b059b9a4659a02ed29/lib-ee/emqx_ee_bridge/src/kafka/emqx_bridge_impl_kafka_producer.erl#L324) for `queuing` and `inflight` after Wolff's upgrade... :see_no_evil:

File: apps/emqx_resource/src/emqx_resource_metrics.erl [hunk 1]
```diff
@@ -152,6 +147,34 @@ handle_telemetry_event(
...
+    end;
+handle_telemetry_event(
+    [?TELEMETRY_PREFIX, Event],
+    _Measurements = #{gauge_shift := Val},
+    _Metadata = #{resource_id := ID, worker_id := WorkerID},
+    _HandlerConfig
+) ->
+    case Event of
+        batching ->
+            emqx_metrics_worker:shift_gauge(?RES_METRICS, ID, WorkerID, 'batching', Val);
```
Comment (zmstone) [inline apps/emqx_resource/src/emqx_resource_metrics.erl:174]: we can delete the batching ?

File: apps/emqx_bridge/i18n/emqx_bridge_schema.conf [hunk 2]
```diff
@@ -195,7 +195,7 @@ emqx_bridge_schema {
 
     metric_sent_inflight {
```
Comment (zmstone) [inline apps/emqx_bridge/i18n/emqx_bridge_schema.conf:196]: nit: inflight is sent
```suggestion
    metric_inflight {
```

File: apps/emqx_resource/src/emqx_resource_metrics.erl (hunk 1 above)
Comment (thalesmg) [reply apps/emqx_resource/src/emqx_resource_metrics.erl:174]: I did it in this other refactoring of the buffer workers: https://github.com/emqx/emqx/pull/9642

**Review by zmstone — APPROVED** [review]

File: apps/emqx_bridge/i18n/emqx_bridge_schema.conf (hunk 2 above)
Comment (thalesmg) [reply apps/emqx_bridge/i18n/emqx_bridge_schema.conf:196]: Will fix it here: https://github.com/emqx/emqx/pull/9642

**Comment by thalesmg** [comment]
The spellcheck failure is unrelated to the changes here and was fixed [elsewhere](https://github.com/emqx/emqx/pull/9658).

**Pull request merged**